lib/resourcemerge/core_test.go

```diff
@@ -16,14 +16,54 @@ func TestEnsurePodSpec(t *testing.T) {
 
 		expectedModified bool
 		expected         corev1.PodSpec
-	}{{
-		name:     "empty inputs",
-		existing: corev1.PodSpec{},
-		input:    corev1.PodSpec{},
-
-		expectedModified: false,
-		expected:         corev1.PodSpec{},
-	}}
+	}{
+		{
+			name:     "empty inputs",
+			existing: corev1.PodSpec{},
+			input:    corev1.PodSpec{},
+
+			expectedModified: false,
+			expected:         corev1.PodSpec{},
+		},
+		{
+			name: "remove regular containers from existing",
+			existing: corev1.PodSpec{
+				Containers: []corev1.Container{
+					corev1.Container{Name: "test"}}},
+			input: corev1.PodSpec{},
+
+			expectedModified: true,
+			expected:         corev1.PodSpec{},
+		},
+		{
+			name: "remove regular containers from existing but dont remove init containers",
```

Comment:
I'm [still][1] not clear on why we want this.  If "modified" means "[launch a replacement Pod][2]", then I'd expect we want to set `modified` and remove entries from `InitContainers` that weren't in the new required PodSpec.  I've floated 8c83b0c9199414128a2e8e63b095be4ec0099639 with an implementation that you can squash into your branch if that makes sense.

[1]: https://github.com/openshift/cluster-version-operator/pull/178#discussion_r291668007
[2]: https://github.com/openshift/cluster-version-operator/pull/178#discussion_r291457813

Reply:
Was trying to play safe and do minimal changes which are required for the issue that i have been trying to fix.
Updated with changes from your commit. Thanks a lot!